lib/model/folder.go

```diff
@@ -182,16 +178,23 @@ func (f *folder) Serve(ctx context.Context) error {
 			return nil
 
 		case <-f.pullScheduled:
-			_, err = f.pull()
-
-		case <-f.pullFailTimer.C:
+			timeutil.StopAndDrain(pullFailTimer)
 			var success bool
+			t0 := time.Now()
 			success, err = f.pull()
 			if (err != nil || !success) && f.pullPause < 60*f.pullBasePause() {
 				// Back off from retrying to pull
 				f.pullPause *= 2
+
+				// Pulling failed, try again later.
+				delay := f.pullPause + time.Since(t0)
+				l.Infof("Folder %v isn't making sync progress - retrying in %v.", f.Description(), stringutil.NiceDurationString(delay))
+				timeutil.ResetTimer(pullFailTimer, delay)
 			}
 
+		case <-pullFailTimer.C:
+			f.SchedulePull()
```

Comment:
Afaik you also need to schedule a pull instead of calling `.pull()` in the next `case`. Resp. maybe it's safer to pass the fail timer into `pull` and do the resetting there, that way one can't call `pull` and forget to handle the timer.